### PR TITLE
build: automate deploy

### PR DESCRIPTION
**Pull request by samsiegart**
fixes https://github.com/Agoric/wallet-app/issues/40

Looking at the whole tree is probably clearer than the diff: https://github.com/Agoric/wallet-app/tree/automate-deploy

See cloudflare deployment working below

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,199 +1,90 @@
...
-    "uuid": "^8.3.2"
+  "private": true,
+  "repository": "https://github.com/Agoric/wallet-app",
+  "useWorkspaces": true,
+  "workspaces": [
+    "wallet"
+  ],
+  "scripts": {
+    "build": "rm -rf build && yarn build:wallet && rollup -c && cp _redirects build",
+    "build:wallet": "rm -rf public/wallet && mkdir public/wallet && yarn workspaces run build && cp -r wallet/build/* public/wallet",
```
Comment (samsiegart) [inline package.json:14]: This is the crux of the change

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -15,9 +15,8 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-node@v3
         with:
-          node-version: '14'
+          node-version: '18.x'
```
Comment (turadg) [inline .github/workflows/test.yml:18]: consider `latest`. the build tools will always work on LTS releases.
https://github.com/actions/setup-node#supported-version-syntax

File: .gitignore [hunk 3]
```diff
@@ -1,24 +1,61 @@
...
-.env.production.local
+.vscode/
+.idea/
 
-npm-debug.log*
-yarn-debug.log*
-yarn-error.log*
+/packages.png
+packages/ui-components/compiled
+packages/ui-components/dist
```
Comment (turadg) [inline .gitignore:59]: any reason not to ignore all `complied` and `dist`?

File: README.md [hunk 4]
```diff
@@ -1,45 +1,15 @@
-# Smart Wallet UI
+# Agoric Wallet Site
 
 ## Development
+See [wallet/README.md](wallet/README.md)
 
-The UI deps and scripts were bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+## Production
+### Manual testing
+`yarn && yarn build && yarn preview`
```
Comment (turadg) [inline README.md:8]: consider `yarn install` for clarity

File: components/install-ses-lockdown.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
+import 'ses';
+import '@agoric/eventual-send/shim';
+
+lockdown({
+  errorTaming: 'unsafe',
```
Comment (turadg) [inline components/install-ses-lockdown.js:6]: worth a comment on why this is okay here

File: package.json [hunk 6]
```diff
@@ -1,199 +1,90 @@
 {
-  "name": "@agoric/wallet-ui",
-  "version": "0.5.0",
-  "description": "Wallet React frontend",
-  "author": "Agoric",
-  "license": "Apache-2.0",
-  "homepage": "/wallet",
+  "name": "@agoric/wallet",
```
Comment (turadg) [inline package.json:2]: more specifically,
```suggestion
  "name": "@agoric/wallet-site",
```

File: package.json [hunk 7]
```diff
@@ -1,199 +1,90 @@
...
-    ]
+  "prettier": {
+    "trailingComma": "all",
+    "singleQuote": true
   },
   "publishConfig": {
     "access": "public"
+  },
+  "engines": {
+    "node": ">=11.0"
```
Comment (turadg) [inline package.json:82]: this is a private package so let's make maintenance easier on ourselves and use the active LTS.
```suggestion
    "node": ">=18.0"
```

File: wallet/.gitignore [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+/node_modules
+/.pnp
+.pnp.js
+
+# testing
+/coverage
+
+# production
+/build
+/public/lockdown.umd.js
```
Comment (turadg) [inline wallet/.gitignore:13]: this one got in. please do remove it and massage gitignore so it doesn't get in again

File: wallet/package.json [hunk 9]
```diff
@@ -0,0 +1,200 @@
+{
+  "name": "@agoric/wallet-ui",
```
Comment (turadg) [inline wallet/package.json:2]: We've also been calling this the "wallet app". wallet.agoric.app is where you get it.

"wallet-ui" or "wallet-app" both work for me but whichever you pick please ensure the workspace (parent path) matches the name here.

File: wallet/src/components/SmartWalletConnection.tsx [hunk 10]
```diff
@@ -40,19 +40,21 @@ export const useProvisionPoolMetrics = (unserializer, leader) => {
 
     let cancelled = false;
     const fetchData = async () => {
-      const follower = await makeFollower(
+      const follower = await makeFollower<{ value: ProvisionPoolMetrics }>(
```
Comment (turadg) [inline wallet/src/components/SmartWalletConnection.tsx:43]: nice

File: wallet/src/util/keyManagement.ts [hunk 11]
```diff
@@ -250,6 +250,7 @@ export const makeBackgroundSigner = async ({
     granter: string,
     rpcClient: import('@cosmjs/tendermint-rpc').Tendermint34Client,
   ): Promise<GenericAuthorization[]> => {
+    // @ts-expect-error
```
Comment (turadg) [inline wallet/src/util/keyManagement.ts:253]: curious what this is. consider some prose explaining the need to suppress.

File: wallet/src/util/querySwingsetParams.ts [hunk 12]
```diff
@@ -18,6 +18,16 @@ export const querySwingsetParams = async (
 ): Promise<QueryParamsResponse> => {
   const http = new HttpClient(endpoint);
   const trpc = await Tendermint34Client.create(http);
+  /**
+   * XXX
```
Comment (turadg) [inline wallet/src/util/querySwingsetParams.ts:22]: oh is this what the other ones are too? consider filing a ticket in this repo to fix that and linking to it at the sites. second thought, probably not worth the overhead.

**Review by turadg — APPROVED** [review]
Glorious!

Please look over the suggestions and requests before merging. Approving now because I'm so excited to see 58 closed  😸

**Comment by samsiegart** [comment]
I tried addressing some feedback but something broke the cloudflare build and I don't have access to the logs. I'll just merge this so we can tweak from a clean slate.

**Pull request merged**